roles/postgresql_upgrade/tasks/main.yml

```diff
@@ -0,0 +1,123 @@
+---
+- name: >-
+    Check if new postgres directory already exists
+  ansible.builtin.stat:
+    path: "{{ postgresql_upgrade_data_dir }}"
+  register: postgresql_upgrade_new_data_dir
+
+- name: >-
+    Fail if new postgres data directory already exists
+  ansible.builtin.fail:
+    msg: >
+      Cannot upgrade because the new postgres directory
+      {{ postgresql_upgrade_data_dir }} already exists; this suggests an
+      upgrade has already been attempted
+  when: postgresql_upgrade_new_data_dir.stat.exists and postgresql_upgrade_new_data_dir.stat.isdir
+
+- name: Back up PostgreSQL
+  ansible.builtin.command: "{{ postgreql_upgrade_backup_script }}"
+  become: true
+  become_user: postgres
+  changed_when: false
+
+- name: Query for all tables that have WITH OIDS set
+  community.postgresql.postgresql_query:
+    db: "{{ postgressql_upgrade_database }}"
+    query: >
+      SELECT 'ALTER TABLE "' || n.nspname || '"."' || c.relname || '" SET WITHOUT OIDS;' AS "query"
+      FROM pg_catalog.pg_class c
+      LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+      WHERE 1=1
+        AND c.relkind = 'r'
+        AND c.relhasoids = true
+        AND n.nspname <> 'pg_catalog'
+      order by n.nspname, c.relname;
+  become: true
+  become_user: postgres
+  register: postgresql_upgrade_tables_with_oids_ddl
+  when: postgresql_upgrade_check_for_oids
+
+- name: Run ALTER statements for tables with OIDS
+  community.postgresql.postgresql_query:
+    db: "{{ postgressql_upgrade_database }}"
+    query: "{{ item.query }}"
+  loop: "{{ postgresql_upgrade_tables_with_oids_ddl.query_result }}"
+  become: true
+  become_user: postgres
+  when: postgresql_upgrade_check_for_oids
+
+- name: Back up PostgreSQL after OIDS edit
+  ansible.builtin.command: "{{ postgreql_upgrade_backup_script }}"
+  become: true
+  become_user: postgres
+  changed_when: false
+  when: postgresql_upgrade_check_for_oids
```

Comment:
I'm not sure but I think this only needs to be done when upgrading to version 12. Perhaps worth keeping though in case we run into old databases that need upgrading?

Reply:
I'm not sure - TBH I thought this was an XNAT thing.